dashboard/app/models/pd/enrollment.rb

```diff
@@ -33,11 +35,30 @@ class Pd::Enrollment < ActiveRecord::Base
   # The enrollment is from one of 2 sources:
   #   1. Web form filled out by user - all school fields required.
   #   2. Automatic association for a user who attends the workshop unenrolled.
-  validate :user_or_school_info_required
-  def user_or_school_info_required
-    return if self.user_id
-    errors.add(:school, 'is required') unless self.school
-    errors.add(:school_type, 'is required') unless self.school_type
+  validate :validate_school_district, unless: :has_user?
+  validates :school, presence: true, unless: :has_user?
+
+  def has_user?
+    self.user_id
+  end
+
+  # Validates the district dropdown.  This list is more verbose than it
+  # needs to be, but correlates to the list of valid configurations given
+  # in https://github.com/code-dot-org/code-dot-org/pull/8624.
+  def validate_school_district
```

Comment:
Not necessarily in this PR, but this method is a perfect candidate for the common model mentioned above if you wanted to consolidate & reuse it.